include/extractor/edge_based_graph_factory.hpp

```diff
@@ -152,14 +157,16 @@ class EdgeBasedGraphFactory
 
     unsigned RenumberEdges();
 
-    std::vector<NBGToEBG> GenerateEdgeExpandedNodes();
+    std::vector<NBGToEBG> GenerateEdgeExpandedNodes(const WayRestrictionMap &way_restriction_map);
 
     void GenerateEdgeExpandedEdges(ScriptingEnvironment &scripting_environment,
                                    const std::string &original_edge_data_filename,
                                    const std::string &turn_lane_data_filename,
                                    const std::string &turn_weight_penalties_filename,
                                    const std::string &turn_duration_penalties_filename,
-                                   const std::string &turn_penalties_index_filename);
+                                   const std::string &turn_penalties_index_filename,
+                                   const RestrictionMap &restriction_map,
+                                   const WayRestrictionMap &way_restriction_map);
```

Comment:
Before they were stored as members and used within. We require the maps to check which turns are allowed. The intersection-generation / handling of via-way restrictions is required when looking at the edge-based-edges to find out which of the turns are valid (`entry_allowed` flag).

I just moved this to a reference parameter to get rid of some pointer usage we did beforehand.